### PR TITLE
make sure that binary or operators aren't the same in different contexts

### PR DESCRIPTION
**Pull request by TyOverby**

### CODE REVIEW

**Comment by TyOverby** [comment]
@AlekseyTs

**Comment by TyOverby** [comment]
@dotnet/roslyn-compiler

File: src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs [hunk 1]
```diff
@@ -14,6 +14,42 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
...
+            var orNodes = tree.GetRoot().DescendantNodes().OfType<BinaryExpressionSyntax>().ToArray();
+            Assert.Equal(orNodes.Length, 2);
+
+            var insideEnumDefinition = semanticModel.GetSymbolInfo(orNodes[0]);
+            var insideMethodBody = semanticModel.GetSymbolInfo(orNodes[1]);
+
+            Assert.False(insideEnumDefinition.IsEmpty);
+            Assert.False(insideMethodBody.IsEmpty);
+
+            Assert.NotEqual(insideEnumDefinition, insideMethodBody);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs]: >Assert.NotEqual(insideEnumDefinition, insideMethodBody); [](start = 12, length = 56)

Please also assert what the symbols are, we usually do this by comparing result of ToTestDisplayString.

Comment (TyOverby) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs]: Done

**Pull request merged**